### PR TITLE
Store initial connect retry value in curl config file

### PR DESCRIPTION
**Pull request by meik99**
# Description

The value from the `feature.dynatrace.com/oneagent-initial-connect-retry-ms` feature flag should be passed to a file instead of an environment variable.
This file is now generated by the standalone runner and mounted by the webhook.

## How can this be tested?

* Create a test-namespace with test apps
* Deploy the operator 
* Apply a Dynakube with the mentioned flag set
* Restart the test-apps as soon as the operator deployment is ready
* The test app pods should have the `curl_options.conf` file mounted to `/var/lib/dynatrace/oneagent/agent/customkeys/curl_options.conf`
* Exec into one of the pods
* The log at `/opt/dynatrace/oneagent-paas/log/nginx/ruxitagent_nginx_myapp-__startup_1.0.log` should contain the line `Applied configuration from file: /var/lib/dynatrace/oneagent/agent/customkeys/curl_options.conf`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

File: src/kubeobjects/query/dynakube.go
Comment (mjgrzybek) [inline src/kubeobjects/query/dynakube.go:23]: Queries shouldn't mutate object, so value should be returned instead of a pointer and used as method receivers.

File: src/kubeobjects/query/dynakube.go
Comment (0sewa0) [inline src/kubeobjects/query/dynakube.go:15]: could be in `dynatracev1beta1` so it can be reused in other places (test and such)

File: src/kubeobjects/query/dynakube.go
Comment (0sewa0) [inline src/kubeobjects/query/dynakube.go:30]: nitpick: pointer receiver is not necessary

Comment (0sewa0) [reply src/kubeobjects/query/dynakube.go:30]: nitpick2: why not pass in the context ? 

I know that our current use of context is "not optimal" but it would make it future proof, and more easily reusable in other parts of the code

File: src/initgeneration/initgeneration.go
Comment (0sewa0) [inline src/initgeneration/initgeneration.go:46]: why remove `ctx`??

I see its not used, but I also see that it could be (here: https://github.com/Dynatrace/dynatrace-operator/pull/806/files#diff-23aafca6480c6cd32914b1b7b9691cbf3acfe3d7ec384c2926b56ebd698c347aR133)

File: src/initgeneration/initgeneration_test.go
Comment (meik99) [reply src/initgeneration/initgeneration_test.go:23]: moved the consts

File: src/kubeobjects/query/dynakube.go
Comment (meik99) [reply src/kubeobjects/query/dynakube.go:15]: moved the consts

File: src/kubeobjects/query/dynakube.go
Comment (meik99) [reply src/kubeobjects/query/dynakube.go:30]: Added option to add context

File: src/initgeneration/initgeneration.go
Comment (meik99) [reply src/initgeneration/initgeneration.go:46]: refactored code to hell and back

**Review by 0sewa0 — DISMISSED** [review]
I really like the refactors

**Comment by meik99** [comment]
I'll wait for @mjgrzybek changes to be merged first before merging

File: src/kubeobjects/query.go
Comment (mjgrzybek) [inline src/kubeobjects/query.go:12]: I didn't find any usage of this member other than passing around.

**Review by mjgrzybek — CHANGES_REQUESTED** [review]

File: src/kubeobjects/query.go
Comment (0sewa0) [reply src/kubeobjects/query.go:12]: Its a leftover, the old `CreateOrUpdateSecret...` used to read the secret, for that it used the `kubeReader`, but the new logic tries to create it and if fails then it will just update it (which I think is simpler and therefore nicer)

but this `kubeQuery` should be reused for other `kubeobject`stuff (daemonset, deployment) and there they might need it

Comment (mjgrzybek) [reply src/kubeobjects/query.go:12]: Agree but it's not tested neither.
To avoid pushing dead code, adding single usage of it would be sufficient.

Comment (meik99) [reply src/kubeobjects/query.go:12]:
```golang
type SecretQuery struct {
	kubeQuery
}

func NewSecretQuery(ctx context.Context, kubeClient client.Client, kubeReader client.Reader, log logr.Logger) SecretQuery {
	return SecretQuery{
		newKubeQuery(ctx, kubeClient, kubeReader, log),
	}
}
```

Comment (meik99) [reply src/kubeobjects/query.go:12]: I found something that could be quickly refactored, give me a minute

Comment (meik99) [reply src/kubeobjects/query.go:12]: done, kubeReader is now used

File: src/kubeobjects/secret.go
Comment (meik99) [reply src/kubeobjects/secret.go:31]: I just do that because otherwise the lines sometimes get long because of the call chains

Comment (mjgrzybek) [reply src/kubeobjects/secret.go:31]: Thus single letter receiver is a pattern.

Comment (meik99) [reply src/kubeobjects/secret.go]: I have been convinced

**Review by mjgrzybek — APPROVED** [review]

**Pull request merged**
